wgengine/wgtun/tun.go

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package wgtun provides a TUN struct implementing the tun.Device interface
+// with additional features as required by wgengine.
+package wgtun
+
+import (
+	"errors"
+	"io"
+	"os"
+	"sync"
+
+	"github.com/tailscale/wireguard-go/device"
+	"github.com/tailscale/wireguard-go/tun"
+	"tailscale.com/types/logger"
+	"tailscale.com/wgengine/filter"
+	"tailscale.com/wgengine/packet"
+)
+
+const (
+	readMaxSize   = device.MaxMessageSize
+	readOffset    = device.MessageTransportHeaderSize
+	MaxPacketSize = device.MaxContentSize
```

Comment:
Exported things should be documented. And not mixed in with unexported consts in the same ( ... ) block.